File: LeetCode/014_LongestCommonPrefix/Solution.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {

        if(strs.length == 0) {
            return "";
        }

        String p = "";
        String s0 = strs[0];

        for(int i = 0; i < s0.length(); i++) {
            char c1 = s0.charAt(i);

            for(int j = 1; j < strs.length; j++) {
                String sj = strs[j];

                if(sj.length() <= i) {
                    return p;
                }
                char c2 = sj.charAt(i);

                if(c1 != c2) {
                    return p;
                }
            }
            p += c1;
        }
        return p;
    }
}

File: LeetCode/014_LongestCommonPrefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    const len = strs.length;
    if (!len) {
        return '';
    }

    const str0 = strs[0];
    const l = str0.length;
    if (l < 1 || !l) {
        return str0;
    }

    let result = '';
    check:
    for (let i = 0; i < l; i++) {
        const s = str0.charAt(i);
        for (let j = 1; j < len; j++) {
            const c = strs[j].charAt(i);
            if (s !== c) {
                break check;
            }
        }
        result += s;
    }
    return result;
};
